postgres-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/integration/postgres/PostgresDatabaseConnector.java

```diff
@@ -23,117 +25,421 @@
 import java.util.ArrayList;
 import java.util.List;
 
-public class PostgresDatabaseConnector extends DatabaseIntegratorConnector {
+public class PostgresDatabaseConnector extends DatabaseIntegratorConnector
+{
 
     @Override
-    public void refresh ( ) throws ConnectorCheckedException {
+    public void refresh() throws ConnectorCheckedException
+    {
         String methodName = "PostgresConnector.refresh";
+        PostgresSourceDatabase sourceDatabase = new PostgresSourceDatabase(connectionProperties);
 
-        //TODO This all needs to be re written so as the controllling loop is through Egeria entities and we then HAVE to as ask postgres if the known state has changed
-        //TODO or is it ok to as for exact database searches such as the below , where only a single entity will be returned
+        try
+        {
+            List<PostgresDatabase> dbs = sourceDatabase.getDabaseNames();
+ /*           List<DatabaseElement> knownDatabases = context.getMyDatabases(1, 100);
+            if( knownDatabases != null )
+            {
+                for (DatabaseElement element : knownDatabases)
+                {
+                    String knownName = element.getDatabaseProperties().getQualifiedName();
+                    for (PostgresDatabase db : dbs)
+                    {
+                        if (db.getQualifiedName().equals(knownName))
+                        {
+                            break;
+                        }
+                        context.removeDatabase(element.getElementHeader().getGUID(), knownName);
+                    }
+                }
+            }
+*/
+            for (PostgresDatabase db : dbs)
+            {
+
+                List<DatabaseElement> database = this.context.getDatabasesByName(db.getQualifiedName(), 0, 100);
```

Comment:
presumably the 0,100 is a pagination limit? current limitation? Add to readme if so? and/or TODO?

Reply:
see it further down too